src/main/java/com/ss/lms/entity/Copies.java

```diff
@@ -1,53 +1,51 @@
 package com.ss.lms.entity;
 
-import java.io.Serializable;
 
-public class Copies implements Serializable {
-
-	private static final long serialVersionUID = -5459027020967892222L;
-	private Integer noOfCopies;
-	private Integer bookId;
-	private Integer branchId;
-	private Book book;
-	private Branch libraryBranch;
-
-	public Book getBook() {
-		return book;
-	}
-
-	public void setBook(Book book) {
-		this.book = book;
-	}
-
-	public Branch getLibraryBranch() {
-		return libraryBranch;
-	}
-
-	public void setLibraryBranch(Branch branch) {
-		this.libraryBranch = branch;
-	}
-
-	public Integer getNoOfCopies() {
+import javax.persistence.Column;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+import org.springframework.stereotype.Component;
+
+@Entity
+@Table(name = "tbl_book_copies")
+@Component
+public class Copies {
+
+
+
+	@EmbeddedId
+	private CopiesId copiesId;
+	
+	@Column(name="noOfCopies")
+	private Long noOfCopies;
+	
+	//@ManyToOne
+	//private Book book;
+	
```

Comment:
doesnt the book and branch relationship have to be mapped or is this portion not completed yet.

Reply:
Yes. Working on it.